checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -159,6 +159,17 @@ protected boolean isInitializationAnnotation(AnnotationMirror anno) {
      */
     public abstract AnnotationMirror getFieldInvariantAnnotation();
 
+    /**
+     * Returns whether or not {@code field} has the invariant annotation.
+     *
+     * <p>If the {@code field} is a type variable, this method returns false if any possible
```

Comment:
Doesn't it return true if any possible instantiation of the type parameter could have the invariant annotation?
Adding a reference to the NullnessATF or an explanation of the check against the lower bound (if the invariant annotation is bottom) would also be helpful.